lib/payment_processor/go_cardless/subscription.rb

```diff
@@ -37,18 +37,19 @@ def initialize(amount:, currency:, user:, page_id:, redirect_flow_id:, session_t
       end
 
       def subscribe
-        subscription = client.subscriptions.create(params: subscription_params)
+        @subscription = client.subscriptions.create(params: subscription_params)
 
-        @local_subscription = Payment::GoCardless.write_subscription(subscription.id, amount_in_whole_currency, currency, @page_id)
         @action = ManageDonation.create(params: action_params)
         @local_customer = Payment::GoCardless.write_customer(customer_id, @action.member_id)
         @local_mandate = Payment::GoCardless.write_mandate(mandate.id, mandate.scheme, mandate.next_possible_charge_date, @local_customer.id)
+        @local_subscription = Payment::GoCardless.write_subscription(@subscription.id, amount_in_whole_currency,
+                                          currency, @page_id, @action.id, @local_customer.id, @local_mandate.id)
       rescue GoCardlessPro::Error => e
         @error = e
       end
 
       def subscription_id
-        @local_subscription.try(:go_cardless_id)
+        @subscription.try(:id)
```

Comment:
`@local_subscription` needs to be created last because it needs to know the ids of the various things created before it. this method gets called when creating the action, so we have to use the id directly from the GC record (created before action), not the local record (created after action)